test/common/upstream/BUILD

```diff
@@ -457,8 +457,8 @@ envoy_cc_test(
 )
 
 envoy_cc_benchmark_binary(
-    name = "load_balancer_benchmark",
-    srcs = ["load_balancer_benchmark.cc"],
+    name = "load_balancer_speed_test",
+    srcs = ["load_balancer_speed_test.cc"],
```

Comment:
The ideal naming for these would be closer to the naming of load_balancer_benchmark and load_balancer_benchmark_test

All the "speed_test" binaries are not actually tests.  The don't use cc_test target.  They are benchmark binaries.  If anything we should rename the speed_tests.

Reply:
I guess my concern is that _benchmark alone is a rather ambiguous.

What about a compromise such as load_balancer_benchmark_test being the invocation (testing) of the benchmark, and give binary we invoke to do that testing a title such as load_balancer_benchmark_utility? E.g. _test in almost all existing envoy cases being the verb/action (and very often synonymous with tool employed to perform that test), while many _test targets actually invoke envoy_static or other scripts or utilities. (For example, build_id_test, envoy_static_test, pie_test, and version_out_test all are actually invoking envoy-static.)

Thoughts?

Reply:
My preference is to not do these cc file renames in this PR.

Ideally the _test suffix should only be used for targets that end up running when you invoke bazel test.  It would be great to use a consistent and unambiguous suffix for all CPU usage measurement targets; I can take an AI to come up with something and make the renames happen.